krm-functions/lib/ipallocation/v1alpha1/ipallocation.go

```diff
@@ -0,0 +1,67 @@
+/*
+  Copyright 2023 Nephio.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	ipamv1alpha1 "github.com/nokia/k8s-ipam/apis/ipam/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/kustomize/kyaml/kio/filters"
+	"sigs.k8s.io/yaml"
+)
+
+
+// IPAllocation interface will refer to all go-functions this library supports
+type IPAllocation interface {
```

Comment:
Are we going to expect multiple struct definitions to implement the IPAllocation interface? Thanks

Reply:
yes we will align this with the interface one. So we will have NewFromYaml and NewFromGostruct.